scripts/variantstore/wdl/GvsExtractCohort.wdl

```diff
@@ -0,0 +1,135 @@
+version 1.0
```

Comment:
It looks to be a little AoU specific, maybe we can use that for the name "GvsAoU..."

Reply:
Setting aside the comments in the WDL that reference AoU specifically.. shouldn't there be alignment on having a general purpose extraction workflow to run all of the extraction steps together, rather than as multiple separate workflows? This is my understanding of the main need that we have (but perhaps there are some other subtleties here). Is it possible to instead generalize anything else here that might be too specific (e.g. make certain tasks skippable/idempotent or generalize naming if needed)?

Reply:
I agree that having a wrapper WDL for the prepare/extract step is a good thing to have.  It's not at the top of our priority list (compared to making sure that it scales, doesn't cost a fortune, etc) but it's a "good thing" for sure.

The reason I suggested naming it AoU specifically was so that we wouldn't hold this up in order to make it generally more useful outside AoU.  If y'all are up for that, that would be great!

There are a few things in here that seem to be AoU specific, for example:
 - requiring a destination bucket to copy the results
 - requiring an extraction uuid
 - not using filter_set (though maybe this is just temporary)

The other thing that I think needs to be figured out is... the lifecycle of the tables created by the Prepare step.  There are temp tables with a TTL, that's probably ok, but then there's the main table used by Extract.  That probably should be cleaned up.   In addition, the creation of the extract table in Prepare is 90% of the cost of this pipeline.  If that succeeds, but then one of the Extract scatters fails... you need to rerun the whole pipeline.  At small scale, you might not care but at larger scale you definitely don't want to do this.  How to best use cromwell call caching, or disable that and do some form of short circuiting in the task if the output already exists, or something else entirely needs to be thought through.

Reply:
I'm also a little hesitant to name it something AoU specific because my intent is for this WDL to be generalized enough for your use cases as well such that we're both working on and using the same WDL. That way, the work we do to improve stuff like table lifecycle and cromwell call caching will automatically be picked up by us when we update the WDL snapshot.     

Given that, all those things you mentioned are definitely changes we can make. Though, I might try to put those changes in a follow up PR so I can wrap this one up soon.

Reply:
I renamed to `GvsExtractCohortFromSampleNames` so it looks a little less like `GvsExtractCallset` and better captures what our intent with the WDL is.

Reply:
I'm all for making this more generally useful if you are -- I'll go make some comments along these lines then.  What I want to avoid is it being really only usable by AoU, yet not saying it's primarily for AoU.